errors on Ubuntu 18.04 with gnu 7.4.0

I'm trying to compile the tests on Ubuntu 18.04 with GNU 7.4.0 and blitz cloned from https://github.com/blitzpp/blitz

I'm getting this error:

```
/home/vagrant/libmpdataxx/tests/unit/../../libmpdata++/kahan_reduction.hpp:13:13: error: expected constructor, destructor, or type conversion before '(' token
 BZ_NAMESPACE(blitz)
             ^
/home/vagrant/libmpdataxx/tests/unit/../../libmpdata++/kahan_reduction.hpp:27:25: error: expected declaration before end of line
 #pragma GCC push_options
                         ^
```
Any ideas?  Here is the singularity file with all the instructions I'm using to create the environment in which I try to compile it:

```
Bootstrap: docker                                                              
From: ubuntu:18.04                                                             
                                                                               
%post                                                                          
export DEBIAN_FRONTEND=noninteractive                                          
export  TZ=America/Los_Angeles                                                 
ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone     
                                                                               
apt-get update -qq \                                                           
    && apt-get install -yq --no-install-recommends \                           
        sudo \                                                                 
        apt-utils \                                                            
        build-essential \                                                      
        pkg-config \                                                           
        git \                                                                  
        cmake \                                                                
        ca-certificates \                                                      
        software-properties-common \                                           
        vim \                                                                  
        curl                                                                   
                                                                               
add-apt-repository ppa:ubuntu-toolchain-r/test                                 
apt-get update                                                                 
                                                                               
add-apt-repository ppa:ubuntu-toolchain-r/test                                 
apt-get update -qq \                                                           
    && apt-get install -yq --no-install-recommends \                           
        wget \                                                                 
        gnuplot-nox \                                                          
        libgnuplot-iostream-dev \                                              
        libhdf5-dev \                                                          
        hdf5-tools \                                                           
        python3-dev \                                                          
        python3-h5py \                                                         
        python3-scipy \                                                        
        python3-matplotlib \                                                   
        libthrust-dev \                                                        
        libboost-all-dev                                                       
                                                                               
git clone --depth=1 git://github.com/thrust/thrust.git                         
sudo ln -s `pwd`/thrust/thrust /usr/local/include/thrust                       
                                                                               
git clone --depth=1 git://github.com/blitzpp/blitz.git                         
cd blitz                                                                       
mkdir build                                                                    
cd build                                                                       
cmake ..                                                                       
make                                                                           
sudo make install                                                              
                                                                               
cd ../../ 
```

--- a/libmpdata++/kahan_reduction.hpp
+++ b/libmpdata++/kahan_reduction.hpp
@@ -10,7 +10,8 @@
 
 // local definition of a Kahan's sum reduction
 // (http://en.wikipedia.org/wiki/Kahan_summation_algorithm)
-BZ_NAMESPACE(blitz)
+namespace blitz
+{
   template<typename P_sourcetype, typename P_resulttype = BZ_SUMTYPE(P_sourcetype)>
   class ReduceKahanSum 
   {
@@ -55,4 +56,4 @@ BZ_NAMESPACE(blitz)
   };
   BZ_DECL_ARRAY_PARTIAL_REDUCE(kahan_sum, ReduceKahanSum)
   BZ_DECL_ARRAY_FULL_REDUCE(kahan_sum, ReduceKahanSum)
-BZ_NAMESPACE_END
+}

